models/tasks.js

```diff
@@ -88,24 +88,74 @@ const addDependency = async (data) => {
  *
  * @return {Promise<tasks|Array>}
  */
+
+const getBuiltTasks = async (tasksSnapshot) => {
+  const tasks = buildTasks(tasksSnapshot);
+  const promises = tasks.map(async (task) => fromFirestoreData(task));
+  const updatedTasks = await Promise.all(promises);
+  const taskPromises = updatedTasks.map(async (task) => {
+    task.status = TASK_STATUS[task.status.toUpperCase()] || task.status;
+    const taskId = task.id;
+    const dependencySnapshot = await dependencyModel.where("taskId", "==", taskId).get();
+    task.dependsOn = [];
+    dependencySnapshot.docs.forEach((doc) => {
+      const dependency = doc.get("dependsOn");
+      task.dependsOn.push(dependency);
+    });
+    return task;
+  });
+  const taskList = await Promise.all(taskPromises);
+  return taskList;
+};
+
+const fetchPaginatedTasks = async ({ status = "", size = TASK_SIZE, page, next, prev }) => {
+  try {
+    const initialQuery = status
+      ? tasksModel.where("status", "==", status).orderBy("title")
+      : tasksModel.orderBy("title");
+    let queryDoc = initialQuery;
+
+    if (prev) {
+      queryDoc = queryDoc.limitToLast(size);
+    } else {
+      queryDoc = queryDoc.limit(size);
+    }
+
+    if (page) {
+      const startAfter = size * page;
+      queryDoc = queryDoc.offset(startAfter);
```

Comment:
Offest?

Reply:
The requirement was to allow paginating even using page number (let's say when not done by the UI). For handling it, used `offset()` from firestore that skips startAfter records.